test/jit/test_backends.py

```diff
@@ -69,47 +76,129 @@ def compare_py_jit_backend(self, name, input):
         pre_compile_spec = self.lowered_module.__getattr__("__method_compile_spec")
 
-        # Load the lowered module.
-        buffer.seek(0)
-        self.lowered_module = torch.jit.load(buffer)
+        # Save and load the lowered module.
+        self.save_load()
 
         # Get the compile spec after loading.
         post_compile_spec = self.lowered_module.__getattr__("__method_compile_spec")
 
         # Compile specs should match.
         self.assertEqual(pre_compile_spec, post_compile_spec)
 
+        # Loaded module should produce the same outputs.
+        self.test_execution()
+
+
+class NestedModuleTest(JitBackendTestCase):
+    """
+    Tests for NestedModule that check that a module lowered to a backend can be used
+    as a submodule.
+    """
+    class NestedModule(torch.nn.Module):
+        """
+        A Module with one submodule that is used to test that lowered Modules
+        can be used as submodules.
+        """
+
+        def __init__(self, submodule):
+            super().__init__()
+            self.submodule = submodule
+
+        def forward(self, x, h):
+            return self.submodule.forward(x, h)
+
+    def setUp(self):
+        super().setUp()
+        # Create Python, JIT and backend versions of NestedModule.
+        # Both modules in self.module are regular Python modules.
+        self.module = NestedModuleTest.NestedModule(BasicModule())
+        # Both modules in self.scripted_module are ScriptModules.
+        self.scripted_module = torch.jit.script(NestedModuleTest.NestedModule(BasicModule()))
+        lowered_module = to_test_backend_multi(
+            self.scripted_module._c, {"forward": {"": ""}}
+        )
+        # self.lowered_module is a ScriptModule, but its submodule is a lowered module.
+        self.lowered_module = torch.jit.script(NestedModuleTest.NestedModule(lowered_module))
+
+    def test_execution(self):
         # Test execution with backend against Python and JIT.
         input = torch.randn(5)
 
-        # Test all three module methods.
-        self.compare_py_jit_backend("accum", input)
-        self.compare_py_jit_backend("sub_accum", input)
-        self.compare_py_jit_backend("forward", input)
+        # Test forward.
+        self.check_function("forward", input)
+
+    def test_save_load(self):
+        # Lowered module should produce the same outputs.
+        self.test_execution()
+
+        # Save and load the lowered module.
+        self.save_load()
+
+        # Loaded module should produce the same outputs.
+        self.test_execution()
+
+
+class TestBackends(JitTestCase):
+    """
+    This class wraps and invokes all subclasses of JitBackendTestCase so that each one
+    does not have to be individually imported in test_jit.py.
+    """
+
+    def __init__(self, name):
+        super().__init__(name)
+        self.basic_module_test = BasicModuleTest()
+        self.nested_module_test = NestedModuleTest()
```

Comment:
You need to construct BasicModuleTest/NestedModuleTest with the same name
```suggestion
        self.basic_module_test = BasicModuleTest(name)
        self.nested_module_test = NestedModuleTest(name)
```